planner/core/rule_column_pruning.go

```diff
@@ -262,8 +262,15 @@ func (p *LogicalTableDual) PruneColumns(parentUsedCols []*expression.Column) err
 		}
 	}
 	for k, cols := range p.schema.TblID2Handle {
-		if p.schema.ColumnIndex(cols[0]) == -1 {
+		for i := len(cols) - 1; i >= 0; i-- {
```

Comment:
The content of `TblID2Handle` looks weird for the above test case, i.e, `t1.a` and `t2.a` are all stored in single bucket of TblID2Handle[70] in my instance for example, though `t1` and `t2` are tow aliases for same table `t`, actually they should be 2 separate tables in the query. Would it have risk in executor when using `TblID2Handle` like this?

Reply:
Is any situation that the handle has not only one column? I think `TblID2Handle` should be the `map<id, *Column>`.

Reply:
Anyway. This is one of the chaos of `Schema`. Maybe temporarily fix bugs is OK.

Reply:
@lzmhhh123 Multi delete will map to multiple Column.